flexboxAPIspecV2.md

```diff
@@ -0,0 +1,783 @@
+         static void SetBasis(Windows.UI.Xaml.UIElement element, Double value);
+         static Double GetBasis(Windows.UI.Xaml.UIElement element);
+    }
+```
+
+
+
+
+## Flex AlignSelf
+
+The align-self property specifies the alignment for the selected item inside the flexible container.
+
+Note: The align-self property overrides the flexible container's align-items property.
+
+```xml
+<controls:LayoutPanel VerticalAlignment="Stretch" MinHeight="100">
+    <controls:LayoutPanel.Layout>
+        <controls:FlexboxLayout Wrap="Wrap" AlignItems="Start"/>
+    </controls:LayoutPanel.Layout>
+        
+            <Grid MinWidth="80"/>
+            <Grid MinWidth="40"/>
+            <Grid MinWidth="100" controls:FlexboxLayout.AlignSelf="End">
+                <TextBlock>End</TextBlock>
+            </Grid>
+            <Grid MinWidth="80"/>
+        </controls:LayoutPanel>
+```
+
+### Output of the above code (styling omitted)
+![image](./FlexAlignSelf.PNG)
+
+##  Flex AlignSelf Accepted Values
+
+| Value | Description |
+|-|-|
+| Auto (Default) |The element inherits its parent container's align-items property, or "stretch" if it has no parent container	|
+| Start | 	The element is positioned at the beginning of the container|
+| End | The element is positioned at the end of the container	|
+| Center | The element is positioned at the center of the container	|
+| Stretch |The element is positioned to fit the container	|
+
+# API Details
+
+```c# (but really MIDL3)
+
+runtimeclass FlexboxLayout;
+
+    enum FlexboxAlignSelf
+        {
+            Auto = 0,
+            Start = 1,
+            End = 2,
+            Center = 3,
+            Stretch = 4,
+        }
+```
+
+
+## Flexbox Unit
+
+The flex-unit property will define the unit of measurement to be used by the flex-basis property.
+
+
+```xml
+        <controls:LayoutPanel MinWidth="500">
+            <controls:LayoutPanel.Layout>
+                <controls:FlexboxLayout/>
+            </controls:LayoutPanel.Layout>
+            <Grid controls:FlexboxLayout.Basis="50" Unit="px">
+                <TextBlock>50</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Basis="100" Unit="px">
+                <TextBlock>100</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Basis="150" Unit="px">
+                <TextBlock>150</TextBlock>
+            </Grid>
+        </controls:LayoutPanel>
+```
+
+### Output of the above code (styling omitted)
+![image](./FlexBasis.PNG)
+
+
+##  Flex AlignSelf Accepted Values
+
+| Value | Description |
+|-|-|
+| px (Default) |The unit of measurment input into the flex-basis property will be calculated in pixels |
+| % | The element's size will be calculated as a relative percentage of its container|
+
+
+
+# API Details
+
+```c# (but really MIDL3)
+
+runtimeclass FlexboxLayout;
+
+    enum FlexboxUnit
+        {
+            px = 0,
+            % = 1,
+
+        }
+```
+
+# Appendix
+
```

Comment:
two open issues:
- *Layout*: how do flexbox layout properties interact (or don't) with XAML layout properties, e.g. `HorizontalAlignment` exists in both
- *Markup*: we'll likely need work in the parser (and some sort of `FlexboxUnitHelper` type) so that the xaml compiler can understand map:
  - "30"  --> { 30, FlexboxUnit::Pixels }
  - "30px" --> { 30, FlexboxUnit::Pixels }
  - "30%" --> { 30, FlexboxUnit::Percent }